packages/store/addon/-private/system/weak-cache.ts

```diff
@@ -0,0 +1,70 @@
+import { DEBUG } from '@glimmer/env';
+
+import { addSymbol, symbol } from '../ts-interfaces/utils/symbol';
+
+interface WeakCache<K extends object, V> {
```

Comment:
Thoughts on a name such as `NamedWeakCache` ?

Reply:
It seems odd to name this differently the `WeakMap`

Reply:
Discussed OOB that since this provides additional cache methods `WeakCache` is a good name.